Client/Game.cpp

```diff
@@ -15,21 +15,27 @@ Material *testMat = nullptr;
 
 void Game::onGameObjectCreated(Connection *c, NetBuffer &buffer) {
 	auto gameObjectType = buffer.read<GAMEOBJECT_TYPES>();
-	GameObject *obj = nullptr;
+	std::unique_ptr<GameObject> obj = nullptr;
 	switch (gameObjectType) {
 		case GAMEOBJECT_TYPES::PLAYER_TYPE:
-			obj = new Player(-1);
+			obj = std::make_unique<Player>(-1);
 			break;
 		default:
 			std::cerr << "Unknown game object type (" << gameObjectType << ")"
 				<< std::endl;
+			return;
 	}
 
 	obj->deserialize(buffer);
-	gameObjects.push_back(obj);
+
+	auto clientObj = new ClientGameObject(std::move(obj));
+	std::cout << clientObj->getGameObject()->getId() << std::endl;
+	// Do not use `obj` after this, ownership transfered to clientObj.
+	clientObj->setModel("Models/sphere.obj");
+	gameObjects[clientObj->getGameObject()->getId()] = clientObj;
 }
 
-Game::Game() {
+Game::Game(): gameObjects(1024, nullptr) {
```

Comment:
What will the real values be for this? Is it supposed to be 1024 and nullptr?

Reply:
`gameObjects` was changed to be a "fixed" length array (maybe we should really change its type to `std::array` instead of `std::vector` to make it actually fixed length) which holds pointers to game objects.

In this case, 1024 is suppose to be the maximum number of game objects in the world, and `nullptr` is just to have the vector initially contain nulls. We can move the 1024 to a constant somewhere, though.

Reply:
Sounds good to me!